Source/WebCore/rendering/StyledMarkedText.cpp

```diff
@@ -136,18 +186,49 @@ Vector<StyledMarkedText> StyledMarkedText::subdivideAndResolve(const Vector<Mark
         return { styledMarkedText };
     }
 
-    auto markedTexts = MarkedText::subdivide(textsToSubdivide);
+    auto markedTexts = MarkedText::subdivide(textsToSubdivide, OverlapStrategy::None);
     ASSERT(!markedTexts.isEmpty());
     if (UNLIKELY(markedTexts.isEmpty()))
         return { };
 
+    bool isHighlights = false;
+
+    if (!markedTexts.isEmpty()) {
+        // Get rid of duplicates in vector, check if contains highlights
+        ListHashSet<MarkedText> noDuplicates;
+        for (auto& markedText : markedTexts) {
+            noDuplicates.add(markedText);
+            if (!isHighlights && markedText.type == MarkedText::Type::Highlight)
+                isHighlights = true;
+        }
+        // Add back to markedTexts
+        if (!noDuplicates.isEmpty()) {
```

Comment:
Nit - I think this `!noDuplicates.isEmpty()` check is unnecessary, since we only get here if `markedTexts` is not empty, so we'll add 1 entry to `noDuplicates` above.